source/common/http/conn_pool_grid.cc

```diff
@@ -299,7 +302,14 @@ ConnectionPool::Cancellable* ConnectivityGrid::newStream(Http::ResponseDecoder&
     createNextPool();
   }
   PoolIterator pool = pools_.begin();
-  if (!shouldAttemptHttp3() || !options.can_use_http3_) {
+  Instance::StreamOptions overriding_options(options);
+  bool delay_tcp_attempt{true};
```

Comment:
nit: we have two initializations, one uses {} and the other (). Could they both use {} (or even = to be more idiomatic)?